src/data_lakehouse_ingest/config_loader.py

```diff
@@ -0,0 +1,216 @@
+        logger: Optional[logging.Logger] = None,
+        minio_client: Optional[Any] = None,
+    ):
+        self.logger = logger or logging.getLogger(__name__)
+        self.minio_client = minio_client
+
+        if isinstance(cfg, str) and cfg.startswith("s3a://") and self.minio_client is None:
+            self.logger.error("❌ MinIO client must be provided for s3a:// paths.")
+            raise ValueError("MinIO client must be provided when loading configuration from an s3a:// path.")
+
+        try:
+            self.config: Dict[str, Any] = self._load_config(cfg)
+            self.logger.info("✅ Config loaded successfully")
+            self._validate_minimal_fields()
+        except Exception as e:
+            self.logger.error(f"❌ ConfigLoader initialization failed: {e}", exc_info=True)
+            raise
+
+    # ----------------------------------------------------------------------
+    # Internal helpers
+    # ----------------------------------------------------------------------
+    def _load_config(self, cfg: Union[str, Dict[str, Any]]) -> Dict[str, Any]:
+        """Load config from dict, local file, MinIO path, or JSON string."""
+        if isinstance(cfg, dict):
+            self.logger.info("📄 Using inline dict configuration")
+            return cfg
+
+        # --- Local file ---
+        if os.path.exists(cfg):
+            safe_base = Path.cwd().resolve()  # or define a fixed config directory
+            requested_path = Path(cfg).resolve()
+
+            # Check that path is inside safe_base
+            if not str(requested_path).startswith(str(safe_base)):
+                self.logger.error(f"❌ Path traversal attempt detected: {cfg}")
+                raise ValueError(f"❌ Path traversal attempt detected: {cfg}")
+
+            self.logger.info(f"📂 Loading configuration from local file: {requested_path}")
+            try:
+                with open(requested_path, "r", encoding="utf-8") as f:
+                    return json.load(f)
+            except Exception as e:
+                self.logger.error(f"❌ Failed to read local config file {requested_path}: {e}", exc_info=True)
+                raise
+
+        # --- MinIO path (s3a://bucket/key) ---
+        if cfg.startswith("s3a://"):
+            path = cfg.replace("s3a://", "")
+            #bucket, key = path.split("/", 1)
+            parts = path.split("/", 1)
+            if len(parts) != 2:
+                raise ValueError(f"Invalid s3a:// path format: {cfg}. Expected s3a://bucket/key")
+            bucket, key = parts
+            self.logger.info(f"📦 Fetching config from MinIO: bucket={bucket}, key={key}")
+            try:
+                response = self.minio_client.get_object(bucket, key)
+                #data = json.loads(response.read())
+                raw_bytes = response.read()
+                data = json.loads(raw_bytes.decode("utf-8"))
+                response.close()
+                response.release_conn()
```

Comment:
Use context manager for safer resource handling - if `response.close()` might not run, if previous code errors.